tests/expectations/metrics/test_core.py

```diff
@@ -99,11 +110,71 @@ def test_mean_metric_pd():
         metrics_to_resolve=(desired_metric,), metrics=metrics
     )
     metrics.update(results)
-    assert results == {desired_metric.id: 2}
+    assert results == {desired_metric.id: result}
 
 
-def test_stdev_metric_pd():
-    engine = build_pandas_engine(pd.DataFrame({"a": [1, 2, 3, None]}))
+@pytest.mark.parametrize(
+    "dataframe,result",
+    [
+        [pd.DataFrame({"a": [1, 2, 3, None]}), 2],
+        [
+            pd.DataFrame({"a": [Decimal(2.0), Decimal(0.18781)]}),
+            1.093905,
+        ],
+    ],
+)
+def test_mean_metric_spark(spark_session, dataframe, result):
+    engine = build_spark_engine(spark=spark_session, df=dataframe, batch_id="my_id")
+
+    metrics: Dict[Tuple[str, str, str], MetricValue] = {}
+
+    table_columns_metric: MetricConfiguration
+    results: Dict[Tuple[str, str, str], MetricValue]
+
+    table_columns_metric, results = get_table_columns_metric(engine=engine)
+    metrics.update(results)
+
+    aggregate_fn_metric = MetricConfiguration(
+        metric_name=f"column.mean.{MetricPartialFunctionTypes.AGGREGATE_FN.metric_suffix}",
+        metric_domain_kwargs={
+            "column": "a",
+        },
+        metric_value_kwargs=None,
+    )
+    aggregate_fn_metric.metric_dependencies = {
+        "table.columns": table_columns_metric,
+    }
+    results = engine.resolve_metrics(metrics_to_resolve=(aggregate_fn_metric,))
+
+    desired_metric = MetricConfiguration(
+        metric_name="column.mean",
+        metric_domain_kwargs={},
+        metric_value_kwargs=None,
+    )
+    desired_metric.metric_dependencies = {
+        "metric_partial_fn": aggregate_fn_metric,
+    }
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(desired_metric,), metrics=results
+    )
+
+    assert results == {desired_metric.id: result}
+
+
+@pytest.mark.parametrize(
+    "build_engine,dataframe,result",
+    [
+        [build_pandas_engine, pd.DataFrame({"a": [1, 2, 3, None]}), 1],
+        # Does not work with Decimals now. MetricResolutionError: unsupported operand type(s) for -: 'float' and 'decimal.Decimal'
```

Comment:
@TrangPham For tidiness purposes, we should clean up before merging (i.e., if we do not need the code currently appearing as commented out, then we should consider deleting it).  Thanks.